src/atlas_f/util/atlas_Geometry_module.F90

```diff
@@ -0,0 +1,179 @@
+
+private :: fckit_owned_object
+
+public :: atlas_Geometry
+
+private
+
+!------------------------------------------------------------------------------
+TYPE, extends(fckit_owned_object) :: atlas_Geometry
+
+! Purpose :
+! -------
+!   *Geometry* : Container of Geometry
+
+! Methods :
+! -------
+
+! Author :
+! ------
+!   April-2020 Benjamin Menetrier     *IRIT-JCSDA*
+
+!------------------------------------------------------------------------------
+contains
+  procedure, public :: delete => atlas_Geometry__delete
+  procedure :: xyz2lonlat => Geometry__xyz2lonlat
+  procedure :: lonlat2xyz => Geometry__lonlat2xyz
+  procedure :: distance_lonlat => Geometry__distance_lonlat
+  procedure :: distance_xyz => Geometry__distance_xyz
+  generic :: distance => distance_lonlat, distance_xyz
+  procedure :: radius => Geometry__radius
+  procedure :: area => Geometry__area
+#if FCKIT_FINAL_NOT_INHERITING
+  final :: atlas_Geometry__final_auto
+#endif
+
+END TYPE atlas_Geometry
+
+!------------------------------------------------------------------------------
+
+interface atlas_Geometry
+  module procedure atlas_Geometry__ctor_name
+  module procedure atlas_Geometry__ctor_radius
+end interface
+
+!------------------------------------------------------------------------------
+
+
+!========================================================
+contains
+!========================================================
+
+
+! -----------------------------------------------------------------------------
+! Geometry routines
+
+function atlas_Geometry__ctor_name(name) result(this)
+  use atlas_Geometry_c_binding
+  use fckit_c_interop_module
+  character(kind=c_char,len=*), intent(in) :: name
+  type(atlas_Geometry) :: this
+  call this%reset_c_ptr( atlas__Geometry__new_name( c_str(name) ) )
+  call this%return()
+end function atlas_Geometry__ctor_name
+
+function atlas_Geometry__ctor_radius(radius) result(this)
+  use atlas_Geometry_c_binding
+  use fckit_c_interop_module
+  real(c_double), intent(in) :: radius
+  type(atlas_Geometry) :: this
+  call this%reset_c_ptr( atlas__Geometry__new_radius( radius ) )
+  call this%return()
+end function atlas_Geometry__ctor_radius
+
+subroutine atlas_Geometry__delete(this)
+  use atlas_Geometry_c_binding
+  class(atlas_Geometry), intent(inout) :: this
+  if ( .not. this%is_null() ) then
+    call atlas__Geometry__delete(this%CPTR_PGIBUG_A)
+  end if
+  call this%reset_c_ptr()
+end subroutine atlas_Geometry__delete
+
+subroutine Geometry__xyz2lonlat(this, x, y, z, lon, lat)
+  use atlas_Geometry_c_binding
+  use fckit_c_interop_module
+  class(atlas_Geometry), intent(in) :: this
+  real(c_double), intent(in) :: x
+  real(c_double), intent(in) :: y
+  real(c_double), intent(in) :: z
+  real(c_double), intent(out) :: lon
+  real(c_double), intent(out) :: lat
+  call atlas__Geometry__xyz2lonlat(this%CPTR_PGIBUG_A, x, y, z, lon, lat)
+end subroutine Geometry__xyz2lonlat
+
+subroutine Geometry__lonlat2xyz(this, lon, lat, x, y, z)
```

Comment:
Can you create another overload geometry%lonlat2xyz( lonlat, xyz )
where xyz is a size(3) array and lonlat a size(2) array ?